### PR TITLE
user_directories in the main config

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- New Feature

Changelog entry:
Add new optional section <user_directories> to the main config.

For example,
```
<user_directories>
    <users_xml>
        <path>/etc/clickhouse-server/users.xml</path>
    </users_xml>
    <local_directory>
        <path>/var/lib/clickhouse/access/</path>
    </local_directory>
</user_directories>
```

The `<user_directories>` section can contain any number of items, the order of the items means their precedence.

### CODE REVIEW

**Comment by qoega** [comment]
Can you look at #13647 before merging your changes?

**Comment by vitlibar** [comment]
> Can you look at #13647 before merging your changes?

I'm not using the `config_dir` parameter in my PR.

**Comment by vitlibar** [comment]
Test [test_materialize_mysql_database/test.py::test_materialize_database_ddl_with_mysql_5_7](https://clickhouse-test-reports.s3.yandex.net/13425/e6c6f904bba6b86b5619ffaaf97412500f8b44d6/integration_tests_(release).html#fail1) might be fixed in https://github.com/ClickHouse/ClickHouse/pull/13822.

File: src/Access/UsersConfigAccessStorage.cpp [hunk 1]
```diff
@@ -493,6 +511,41 @@ void UsersConfigAccessStorage::setConfiguration(const Poco::Util::AbstractConfig
     memory_storage.setAll(all_entities);
 }
 
+void UsersConfigAccessStorage::load(const String & users_config_path,
+          const String & include_from_path,
+          const String & preprocessed_dir,
+          const zkutil::GetZooKeeper & get_zookeeper_function)
+{
+    std::lock_guard lock{load_mutex};
+    path = std::filesystem::canonical(users_config_path);
```
Comment (azat) [inline src/Access/UsersConfigAccessStorage.cpp:520]: This breaks #13515

Comment (vitlibar) [reply src/Access/UsersConfigAccessStorage.cpp:520]: I'll fix that.

**Pull request merged**